src/main/java/com/thebois/models/world/Grass.java

```diff
@@ -35,4 +35,10 @@ public TerrainType getType() {
     public Grass deepClone() {
         return new Grass(getPosition());
     }
+
+    @Override
+    public float getCost() {
+        return 0;
+    }
```

Comment:
How wrong would it be to move this up to `Abstract Terrain`, and let specific terrrain overried this if it should be something else?

Reply:
😂 it was that way at first but @jonathan-lindqvist didn't like that so I refactored it like this.

The benefit of doing it the way you and I thought is that you don't have to implement it in every subtype if they all return 0 anyways. 

The downside to that is when you forget to override the method in a subtype that actually has a bigger cost, you won't know until you run the game and try to get a pawn to pathfind over the tile to see if it avoids it or not. 

So the question is, do we think we'll have more tiles with 0 cost, or more with specified costs?

Reply:
On second thought, you're probably right that most, if not all, terrain will have a movement cost of 0. Then we should probably keep it with a default value in the abstract class and have types override it when needed.

Same goes for structures, most, if not all, will not be walkable at all.

When implementing resources, they should all probably either be unwalkable or very slowly walkable. 

@jonathan-lindqvist thoughts?

Reply:
I agree with this.